activerecord/test/cases/schema_dumper_test.rb

```diff
@@ -3,14 +3,36 @@
 require "cases/helper"
 require "support/schema_dumping_helper"
 
-class SchemaDumperTest < ActiveRecord::TestCase
-  include SchemaDumpingHelper
-  self.use_transactional_tests = false
+if current_adapter?(:SQLite3Adapter) && !in_memory_db?
+  class SchemaDumperWithMigrationsPathsTestCase < ActiveRecord::TestCase
+    include SchemaDumpingHelper
+    self.use_transactional_tests = false
+
+    # Use a memory db here to avoid having to rollback at the end
+    setup do
```

Comment:
To be honest I don't really understand this code, but I copied it from https://github.com/rails/rails/blob/master/activerecord/test/cases/tasks/database_tasks_test.rb#L885..L905 and it does what I need.

What I want to do here is have `ActiveRecord::Base.connection` return a connection which has the `migrations_path` config I provide on line 16. I need this because my tests need to compare what's in the filesystem (at `MIGRATIONS_ROOT + "/schema_dumper"`) and what's in the db (done per-test). I'm open to better/simpler ways of doing this.